### DISCUSSION TITLE
What is the point of org.jobrunr.job-scheduler.enabled property ?

### DISCUSSION THREAD

**Discussion by Mumeii** [jobrunr/jobrunr#522] [Q&A] [answered]
Hi

I'm POCing on JobRunR recently 🥳  (using Spring Boot + Postgresql flavor)

One of my concerns is having the possibility to modularize it : 
- One pod that expose the dashboard UI & manage to bootstrap our batch definition.
- Several other ones that host Background Job Servers to manage the Job requests

I've notice there is (at least) three properties to enable / disable certain parts of JobRunR : 
- `org.jobrunr.background-job-server.enabled` : seems reasonable to set it to false for my first kind of pod (UI & bootstrap one)
- `org.jobrunr.dashboard.enabled` : seems logical to disable the dashboard on pods hosting Background Job Servers, I.E. no need to have them exposing a UI on top of the jobs they'll have to process
- `org.jobrunr.job-scheduler.enabled` : but regarding this parameter, I'm puzzled. Can't tell which effect it exactly has.

From what I've seen during my tests it seems to, at least, prevent the creation of the `JobRequestScheduler` bean, which itself seems to simply give access to the storage part, whichever it is, in order to fill the jobrunr_recurring_jobs & jobrunr_jobs tables

So from my point of view, setting this parameter to false doesn't seems to prevent starting an "active" component, like 
- the dashboard one does by stoping the UI exposition
- the background-job-server one does by preventing the startup of the endless loop that poll on the database content.

That's why I'd like to have your point of view about it, I feel like I'm missing an important point regarding this `job-scheduler.enabled` parameter.

Thanks in advance for your feedback about it !

**Comment by rdehuyss** [comment, marked as answer]
Well, you have 3 important parts in JobRunr:
- scheduling & creating jobs
- processing jobs
- showing the dashboard

Each one of the 3 properties you mentioned, controls one of the aspects above. When I think of it now, the setting may be pre-emptive optimisation. I also don't see a reason why one would disable it. Back then, it made sense to have control over the 3 different parts.

But, your requirements can easily be met using the other 2 properties.

**Reply by Mumeii** [reply]
In fact at first, due to the availability of this property, I tought that the (active) process of turning entries from jobrunr_recurring_jobs into entries in jobrunr_jobs was actionned by this flag.

So I wrongly pictured that I could configure the background servers by enabling `background-job-server.enabled` and disabling `job-scheduler.enabled`

And by extension I also tought wrongly that I would have to configure a third kind of pod with `job-scheduler.enabled`, that would be in charge of ... well ...
scheduling :p

So from my point of view, keeping it could misslead a few newcomers like me.

**Reply by rdehuyss** [reply]
I understand - I'll see whether I can remove it in JobRunr 6.
